### PR TITLE
Auto processor fix

### PR DESCRIPTION
**Pull request by LysandreJik**
Reverts the `AutoProcessor` magic.

### CODE REVIEW

**Comment by LysandreJik** [comment]
I'm OOO on Monday/Tuesday so feel free to merge this so that it doesn't block Wednesday's release.

File: src/transformers/models/auto/processing_auto.py [hunk 1]
```diff
@@ -41,7 +39,6 @@
         ("speech_to_text_2", "Speech2Text2Processor"),
         ("trocr", "TrOCRProcessor"),
         ("wav2vec2", "Wav2Vec2Processor"),
-        ("vision-text-dual-encoder", "VisionTextDualEncoderProcessor"),
```
Comment (sgugger) [inline src/transformers/models/auto/processing_auto.py:44]: No this one should stay here.

File: src/transformers/models/auto/processing_auto.py [hunk 2]
```diff
@@ -144,47 +138,28 @@ def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
...
-        # get_list_of_files only takes three of the kwargs we have, so we filter them.
-        get_list_of_files_kwargs = {
-            key: kwargs[key] for key in ["revision", "use_auth_token", "local_files_only"] if key in kwargs
-        }
-        model_files = get_list_of_files(pretrained_model_name_or_path, **get_list_of_files_kwargs)
-        if FEATURE_EXTRACTOR_NAME in model_files:
-            config_dict, _ = FeatureExtractionMixin.get_feature_extractor_dict(pretrained_model_name_or_path, **kwargs)
-            if "processor_class" in config_dict:
-                processor_class = processor_class_from_name(config_dict["processor_class"])
-                return processor_class.from_pretrained(pretrained_model_name_or_path, **kwargs)
```
Comment (sgugger) [inline src/transformers/models/auto/processing_auto.py:157]: I think it's cleaner to leave this here and only check the feature extractor config for a processor class if there is one available. There might be processors in the future that require no feature extractor.

File: src/transformers/models/auto/processing_auto.py [hunk 3]
```diff
@@ -144,47 +138,28 @@ def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
...
+            return processor_class.from_pretrained(pretrained_model_name_or_path, **kwargs)
 
         # Otherwise, load config, if it can be loaded.
         if not isinstance(config, PretrainedConfig):
             config = AutoConfig.from_pretrained(pretrained_model_name_or_path, **kwargs)
 
         model_type = config_class_to_model_type(type(config).__name__)
 
-        if getattr(config, "processor_class", None) is not None:
+        if getattr(config_dict, "processor_class", None) is not None:
```
Comment (sgugger) [inline src/transformers/models/auto/processing_auto.py:153]:
```suggestion
        if getattr(config, "processor_class", None) is not None:
```
Should be the config.

**Review by sgugger — COMMENTED** [review]
Thanks for taking care of that. Will push on your PR tomorrow the fixes and merge.

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**